### ISSUE TITLE
Calibration and ZP used in GCR

### ISSUE THREAD

**Issue by Manal-Yassine** [LSSTDESC/gcr-catalogs#380]
Hi, Using the calibration (with getPhotoCalib) in the r band to get the magnitude of a src in the difference image (with butler), I found the same results of GCR DM measurement presented in the dc2_dia_object_run1.2p_test (following MWV notebook). Howerver, if I use the calibration for other filters, I found that my result is delayed wrt GCR DM measurement. So it seems that GCR catalog does not take into account the calibration which corresponds to each filter. I remarked as well that the magnitudes for all visits of a diaObject are calculated with one zeropoint (ZP) in a defined filter, so the GCR catalog does not take into account the ZP of the difference image studied in a given filter. To have more details please take a look at my presentation: https://drive.google.com/file/d/1OaKDLbyEGAGPawjEwv_5h0UMnotyOc-1/view

**Comment by wmwv** [comment]
@Manal-Yassine 
Thanks for filing an issue to track this.  There's clearly something wrong at any of the `dia_pipe`, the `DC2-production/scripts/merge_source_cat.py`, or `gcr-catalog` stages.
My apologies that I wasn't able to attend the telecon.

Do you have a specific suggestion for what is wrong that should be fixed?

> So it seems that GCR catalog does not take into account the calibration which corresponds to each filter.
I read through the slides and I see this in the conclusions, but I didn't follow exactly what the evidence was that the GCR catalog is incorrectly using the same calibration for different filters.  I saw that in r, your calculation and the GCR result agreed, and that in other bands it disagreed.  Why should we then conclude that the reason for the disagreement is that the GCR result is based on incorrectly using the same zeropoint across filters?

**Comment by wmwv** [comment]
For some technical details, the GCR result is based on the calibration using the `fluxmag0` from `calib.getFluxMag0()`.

**Comment by Manal-Yassine** [comment]
Hi @wmwv , I guess in gcr-catalog, but I did not go deeply into the details of the code. I maybe not help to precise which script of the code should be modified but I can describe what I did to display DM measurements using Calib.
I tried first of all to retrieve the results of GCR using diaSrc = dc2_dia_source_run1.2p_test.
For a diasrc associated with diaobj: "diaObject_diaSrc" I found its imagediff (the difference image) in a given filter
Then I applied:
calibdiff = imagediff.getPhotoCalib()
this_filter = diaObject_diaSrc.query(f'filter == "{filt}"') # filt is one of the 6 filters 
PsFlux= this_filter['psFlux'].to_numpy()
for i in range(len(PsFlux)):
	MagCalib= calibdiff.instFluxToMagnitude(PsFlux[i])  
using these few lines  with an imagediff obtained in the filter 'r', I obtain the same results as what GCR gave directly (similar to what you show in your notebook sn-dia). However, if I change the imagediff filter, my results become shifted with respect to GCR. On the other hand, as it is clear from the lines above, the same imagediff (this means the same ZeroPoint) is used for all PsFlux list. 
So what I did, is to search for each imagediff the flux of the diasrc (with butler) and use calibdiff for it taking into account the ZP and the filter that I am studying. 
Hope this can help you to know where this should be included in the code,
Thank you
